### PR TITLE
Upgrade Soroban limits instead of using configuration parameters

### PR DESCRIPTION
**Pull request by Shaptic**
### What
This removes the usage of Core configuration flags related to testing, namely:

```toml
TESTING_MINIMUM_PERSISTENT_ENTRY_LIFETIME=10
TESTING_SOROBAN_HIGH_LIMIT_OVERRIDE=true
```

in favor of actually upgrading network limits via Core (see `upgradeLimits`). The settings (both `testnet` and `unlimited`) explicitly set the `min_persistent_ttl` setting to 10. By default, it will set unlimited limits, but we need limits to be enforced for a pending test case.

### Why
We need to enforce limits for the edge case of autorestore where the restore+invoke transaction exceeds limits and needs to be separated into restore then invoke (pending) as part of #464.

### Known limitations
n/a

### CODE REVIEW

**Comment by Shaptic** [comment]
```
running command stellar [contract build] 

stellar: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.39' not found (required by stellar)
stellar: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found (required by stellar)
stellar: /lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.15' not found (required by stellar)
```

This is some weird CI bs that's gonna take a year and a day to investigate, so we'll just have to let it slide for now as this is higher priority.

File: cmd/stellar-rpc/internal/integrationtest/migrate_test.go [hunk 1]
```diff
@@ -14,10 +14,14 @@ import (
...
-// Test that every Stellar RPC version (within the current protocol) can migrate cleanly to the current version
-// We cannot test prior protocol versions since the Transaction XDR used for the test could be incompatible
+// Test that every Stellar RPC version (within the current protocol) can migrate
+// cleanly to the current version. We cannot test prior protocol versions since
+// the Transaction XDR used for the test could be incompatible
+//
 // TODO: find a way to test migrations between protocols
 func TestMigrate(t *testing.T) {
+	t.Skip("Idk why it doesn't work but lemme check others")
+
```
Comment (tamirms) [inline cmd/stellar-rpc/internal/integrationtest/migrate_test.go:24]: it looks like this still needs to be resolved

File: cmd/stellar-rpc/internal/xdr2json/conversion.go [hunk 2]
```diff
@@ -63,16 +63,16 @@ func convertAnyBytes(xdrTypeName string, field []byte) (json.RawMessage, error)
 	// scope just added to show matching alloc/frees
 	{
```
Comment (tamirms) [inline cmd/stellar-rpc/internal/xdr2json/conversion.go:64]: I think the scope is not necessary anymore now that you're using defer to free the memory

File: cmd/stellar-rpc/internal/integrationtest/infrastructure/test.go [hunk 3]
```diff
@@ -719,13 +756,141 @@ func (i *Test) CreateHelloWorldContract() (protocol.GetTransactionResponse, [32]
...
 func (i *Test) InvokeHostFunc(
 	contractID xdr.ContractId, method string, args ...xdr.ScVal,
 ) protocol.GetTransactionResponse {
 	op := CreateInvokeHostOperation(i.MasterAccount().GetAccountID(), contractID, method, args...)
 	return i.PreflightAndSendMasterOperation(op)
 }
 
+//nolint:funlen // it's a test bro relax
+func (i *Test) upgradeLimits() {
+	helper := func(limitFile string) string {
```
[the remark this answers is not in the document]
Comment (Shaptic) [reply cmd/stellar-rpc/internal/integrationtest/infrastructure/test.go]: laziness :laughing: fixed!

File: cmd/stellar-rpc/internal/integrationtest/infrastructure/docker/upgrades/enable.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
```
[the remark this answers is not in the document]
Comment (Shaptic) [reply cmd/stellar-rpc/internal/integrationtest/infrastructure/docker/upgrades/enable.json:1]: brilliant! done

File: cmd/stellar-rpc/internal/integrationtest/migrate_test.go (hunk 1 above)
Comment (Shaptic) [reply cmd/stellar-rpc/internal/integrationtest/migrate_test.go]: Fixed it: when we restart RPC as part of a test (e.g. migration), we need to be sure to (a) not apply a settings upgrade again and (b) not use the Core config overrides since they're incompatible with doing the upgrade "for real."
